rickchurch/auth.py

```diff
@@ -103,13 +115,29 @@ async def reset_user_token(user_id: int, asyncpg_conn: asyncpg.Connection) -> st
     return token
 
 
-def make_user_token(user_id: int) -> Tuple[str, str]:
+async def add_user(user: dict, db_conn: asyncpg.Connection) -> str:
     """
-    Generate a JWT token for given user_id.
+    Add a new church user. If given member already exists,
+    reset his token unless he is banned.
 
-    Returns a tuple of the JWT token, and the token_salt.
+    Return the new user's token.
     """
-    # 22 long string
-    token_salt = secrets.token_urlsafe(16)
-    jwt_data = dict(id=user_id, salt=token_salt)
-    return jwt.encode(jwt_data, constants.jwt_secret, algorithm="HS256"), token_salt
+    user_name = ""  # TODO: find the key for discord username, if not present, request it from discord
```

Comment:
I'm not sure under which key the username is present, if at all, and since the API isn't tested out yet, I won't find that out until it will be. For now leave all usernames as `""`, but it is important that this isn't forgotten and will get fixed in the future

Reply:
Resolved, I verified the request manually, and it returned the username in under "username" key, together with discriminator (4 numbers after name) under "discriminator"